resilience4j-kotlin/src/main/kotlin/io/github/resilience4j/kotlin/timelimiter/TimeLimiter.kt

```diff
@@ -45,9 +46,9 @@ suspend fun <T> TimeLimiter.executeSuspendFunction(block: suspend () -> T): T =
         }
     } catch (t: Throwable) {
         if (isCancellation(coroutineContext, t)) {
-            onError(t)
+            onError(TimeoutException(t.message))
```

Comment:
Would it even make sense to rethrow a  `TimeoutException`? 
It might solve the following issue -> https://github.com/resilience4j/resilience4j/issues/1123

Reply:
The nesting part in Kotlin is indeed a problem. It's something that I solved locally in my application as not to mess the timeLimiter error signals, which, as the comments in the code suggest actually throw a coroutine `TimeoutCancellationException`.

Catching the exception and mapping it to a `java.util.concurrent.TimeoutException` does change the behaviour, and people currently handling CancellationExceptions should be made aware of this change.

Nevertheless, I am in favor of this change, which would also allow me to get rid of the helper function that I wrote here: https://github.com/resilience4j/resilience4j/issues/1168#issuecomment-700704364

Reply:
Do you also have a solution for https://github.com/resilience4j/resilience4j/issues/688

Reply:
Yes, we have to document this behavior in the JavaDoc.
In general this problem is related to issue https://github.com/resilience4j/resilience4j/issues/758